core/modules/wdb/QueryRunner.cc

```diff
@@ -445,6 +441,11 @@ bool QueryRunner::_dispatchChannel() {
     // or there are no more rows to collect, pass _transmitData to _sendChannel.
     try {
         _initTransmit(); // set _transmit
+        ChunkResourceRequest req(_chunkResourceMgr, tMsg);
+        ChunkResource cr(req.getResourceFragment(fragNum));
```

Comment:
These two lines were not present in the older version of the code. And I'm not seeing any use for `cr` in the rest of the method. What's the purpose of these statements?

Reply:
They were there. They have always been there. Exactly where, I'd have to look. I moved them into the try block from line 435, but I don't remeber where they were before that.

These 2 lines create the temporary subchunk tables need for this task. Eventually, through a maze of code, they call the load function that creates the database and in memory tables that we have been discussing in another ticket the past couple of weeks.

The TODO here has to do with the fact that any other query in this chunk, using the subchunks, should be able to reuse these tables, but these tables only last until the end of the try block. They should really last until thes scheduler is done with all the tasks in this chunk, which is a bit tricky to do.